internal/api/front/perses/src/app/project/dashboard/dashboard-details/dashboard-details.component.ts

```diff
@@ -24,30 +29,75 @@ import { DashboardModel } from '../dashboard.model';
   templateUrl: './dashboard-details.component.html',
   styleUrls: ['./dashboard-details.component.scss']
 })
-export class DashboardDetailsComponent implements OnInit {
+export class DashboardDetailsComponent {
+
+  name: string = "";
   isLoading = false;
-  currentProject = '';
   dashboard: DashboardModel = {} as DashboardModel;
+  ngxData = new Map<string, NgxLineChartModel[]>();
+  currentProject = '';
 
-  constructor(private dashboardService: DashboardService,
+  // TEMP static options data for graph display
+  view: [number, number] = [600, 400];
+  showXAxis = true;
+  showYAxis = true;
+  gradient = false;
+  showLegend = true;
+  showXAxisLabel = true;
+  xAxisLabel = 'Time';
+  showYAxisLabel = true;
+  yAxisLabel = 'Value';
+  timeline = true;
+  colorScheme = {
+    domain: ['#9370DB', '#87CEFA', '#FA8072', '#FF7F50', '#90EE90', '#9370DB']
+  };
+
+  constructor(private service: DashboardService,
+              private feedService: DashboardFeedService,
               private readonly toastService: ToastService,
-              private readonly projectService: ProjectService) {
+              private readonly projectService: ProjectService,
+              private route: ActivatedRoute) {
+  }
+
+  ngOnInit(): void {
+    this.isLoading = true;
+
+    this.route.params.subscribe(params => {
+       this.name = params['dashboard'];
+    });
+
     this.projectService.getCurrent().pipe(untilDestroyed(this)).subscribe(
-      current => {
-        this.currentProject = current;
-        this.getDashboard();
+      res => {
+        this.currentProject = res;
+
+        if (Object.keys(this.service.currentDashboard).length != 0
+        && this.service.currentDashboard.metadata.name == this.name) {
+          this.dashboard = this.service.currentDashboard;
+          this.feedDashboard();
```

Comment:
If feedDashboard and getDashboard was returning Observables you could be able to chain them and then have the success / error handling at the same place 
It would simplify for example the management of the isLoading attribute. If you want we can speak about that together.

Reply:
yes, and btw I'm not very proud of this if haha, sharing the data using the service (this.service.currentDashboard) was a quick win solution to achieve what I wanted, but probably this should be revamped completely (stateless approach somehow ?).

I get your idea, 'will try to do sth and we can discuss if it's still not clean afterwards :D